build-tools/packages/build-tools/src/typeValidator/packageJson.ts

```diff
@@ -131,13 +145,155 @@ export async function getPackageDetails(packageDir: string): Promise<PackageDeta
 export type PreviousVersionStyle =
     | "baseMajor"
     | "baseMinor"
+    | "~baseMinor"
+    | "previousPatch"
     | "previousMinor"
     | "previousMajor"
     | "^previousMajor"
     | "^previousMinor"
     | "~previousMajor"
     | "~previousMinor";
 
+/**
+ * Calculates the correct version baseline to use for typetests based on the {@link PreviousVersionStyle}.
+ *
+ * @param version - The version.
+ * @param style - The version style to calculate.
+ * @returns A valid semver version range for the previous version.
+ */
+function getPreviousVersionBaseline(version: ReleaseVersion, style: PreviousVersionStyle): string {
+    const [previousMajorVersion, previousMinorVersion, previousPatchVersion] =
+        getPreviousVersions(version);
+    let prevVersion: string;
```

Comment:
Doesn't look like there are any drop-through cases in the switch below. Would it make sense to just have each case return, rather than setting this and returning at the end?

Reply:
Is there a perf benefit? I find early return-heavy code more difficult to read and refactor (e.g. we later need to do some cleanup on the value before returning it), but I don't understand JS perf very well.